JoomGalleryfriends/JG4-dev
# Checkout/Checkin functionality

## What is this pull request for

Adds the checkout and checkin functionality to image, category, config and tags manager as you know it from com_content. If a user is editing a record, the record gets a lock in the list view. This record is now locked for anybody else to edit until it is unlocked again.

## How to test this  pull request

- Edit a record by opening a second tab (image, category, config set or tag)
- Close tab without closing the edit form with the proper button
- Refresh the list view. The edited record should have now a lock in front of the title.
- If you have the correct permission you can unlock the record byclicking on the lock.
